docs-2.0/4.deployment-and-installation/6.deploy-text-based-index/2.deploy.md

```diff
@@ -0,0 +1,95 @@
+
+```json
+{
+ "template": "nebula*",
+  "settings": {
+    "index": {
+      "number_of_shards": 3,
+      "number_of_replicas": 1
+    }
+  },
+  "mappings": {
+    "properties" : {
+            "tag_id" : { "type" : "long" },
+            "column_id" : { "type" : "text" },
+            "value" :{ "type" : "text"}
+        }
+  }
+}
+```
+
+Make sure that you specify the following fields in strict accordance with the preceding template format:
+
+```text
+"template": "nebula*"
+"tag_id" : { "type" : "long" },
+"column_id" : { "type" : "text" },
+"value" :{ "type" : "text"}
+```
+
+## Deploy full-text cluster
+
+Full-text index data is wrote to the Elasticsearch cluster asynchronously. To use the full-text index search, you must run the full-text storage services as Listeners. Configure the `listener_path` parameter for the Listeners. For example:
+
+```conf
+########## basics ##########
+# The file to host the process id
+--pid_file=pids_listener/nebula-storaged.pid
+```
+
+Add the preceding configurations to the `nebula-graphd.conf` configuration file (the directory is `/usr/local/nebula/etc/` by default) for the Listeners.
+
+## Sign in and sign out to the Elasticsearch clients
+
+```ngql
+SIGN {IN | OUT} TEXT SERVICE [(<ip:port> [,<username>, <password>]), (<ip:port>), ...]
+```
+
+When the Elasticsearch cluster is deployed, use the `SIGN IN | OUT` statement to sign in or sing out to the Elasticsearch clients. Multiple `ip:port` pairs are separated with commas. You must use the real IPs. For example:
+
+```ngql
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200);
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200), (192.168.8.2:9300);
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user", "password");
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user", "password");, (127.0.0.1:9200, "user", "password");
+nebula> SIGN OUT TEXT SERVICE;
+nebula> SIGN IN TEXT SERVICE (192.168.8.1:9200, "user");
+```
+
+## Add and remove full-text Listeners
+
+```ngql
+{ADD | REMOVE} LISTENER ELASTICSEARCH (<ip:port>) [,(<ip:port>), ...]
+```
+
+When Listeners are started, use the `ADD | REMOVE LISTENER ELASTICSEARCH` statement to add or remove Listeners for a graph space. Multiple `ip:port` pairs are separated with commas. You must use the real IPs. For example:
+
+```ngql
+nebula> ADD LISTENER ELASTICSEARCH 192.168.8.210:56789;
+```
+
+## Operate Listeners
+
+```ngql
+{ADD | SHOW | DROP | LIST} LISTENER
+```
+
+Use the preceding statements to add, show, drop, and list the Listeners.
+
+## Full-text index limitations
+
+For now, full-text index has the following limitations:
+
+- Full-text indexes are not deleted together with the graph space.
+- Full-text indexes does not support rebuilding or altering.
+- Full-text indexes does not support indexing multiple properties at one time.
+- Make sure that you start the Elasticserch cluster and Nebula Graph at the same time. Or the data writing on the Elasticserch cluster can be incomplete.
```

Comment:
how about give one or two full-text index examples?  ( it can be an individual .md file)
one for sign in case ( from the very begin ) , and do some lookup
one for sign out
one for some bad case (maybe)